File: WEEK 1/DAY 1 - Array Basics/SOLUTIONS/02_Medium_Problem_Solution/Solution.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        max_sum = nums[0]
        current_sum = 0
        
        for num in nums:
            current_sum = max(num, current_sum + num)
            max_sum = max(max_sum, current_sum)
        
        return max_sum

File: WEEK 1/DAY 1 - Array Basics/SOLUTIONS/02_Medium_Problem_Solution/Solution.java
class Solution {
    public int maxSubArray(int[] nums) {
        int max_sum = nums[0];
        int current_sum = 0;
        
        for (int num : nums) {
            current_sum = Math.max(num, current_sum + num);
            max_sum = Math.max(max_sum, current_sum);
        }
        
        return max_sum;
    }
}
